clang/lib/Parse/ParseStmt.cpp

```diff
@@ -496,9 +496,28 @@ StmtResult Parser::ParseExprStatement(ParsedStmtContext StmtCtx) {
     return ParseCaseStatement(StmtCtx, /*MissingCase=*/true, Expr);
   }
 
+  // Parse where clause on an ExprStmt, if it exists.
+  WhereClause *WClause = nullptr;
+  if (getLangOpts().CheckedC && Tok.is(tok::kw__Where)) {
+    WClause = ParseWhereClause();
+    if (!WClause)
+      return StmtError();
+  }
+
   // Otherwise, eat the semicolon.
   ExpectAndConsumeSemi(diag::err_expected_semi_after_expr);
-  return handleExprStmt(Expr, StmtCtx);
+  StmtResult StmtRes = handleExprStmt(Expr, StmtCtx);
+  if (StmtRes.isInvalid() || !WClause)
```

Comment:
The check `!WClause` may not be necessary - it has already been checked above.

Reply:
The `!WClause` check above is inside `if (getLangOpts().CheckedC && Tok.is(tok::kw__Where))`. If we are not in a where clause then we don't want the rest of the code below to execute. That's why I have this check here.

To make the logic cleaner, we can also have a dedicated variable which checks if where clause is present or not.

Reply:
Yes, thank you for the clarification!